ooyala_player/ooyala_player.py

```diff
@@ -316,7 +316,7 @@ def api_key_3play_with_default_setting(self):
 
         settings_service = self.runtime.service(self, 'settings')
         try:
-            return settings_service.get('ENV_TOKENS')['XBLOCK_OOYALA_3PLAY_API']
+            return settings_service.get_settings_bucket(self, {'3PLAY_API_KEY': self.api_key_3play})['3PLAY_API_KEY']
         except (AttributeError, KeyError):
             return self.api_key_3play
```

Comment:
+1 on Braden's comment.  More specifically, line 319 should simply read

    return settings_service.get_settings_bucket(self).get('3PLAY_API_KEY')

since (a) `get_settings_bucket()` returns an empty dictionary by default and (b) even if you get back an actual settings bucket, it's not guaranteed that it contains the key `3PLAY_API_KEY`.

You could also replace the try/except block by

    if settings_service is not None:
        return settings_service.get_settings_bucket(self).get('3PLAY_API_KEY')
    else:
        return None

to be a bit more explicit than just saying that there might be random AtrributeErrors and KeyErrors.